server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/UnsignedLongValuesComparatorSource.java

```diff
@@ -78,7 +78,7 @@ private NumericDocValues getNumericDocValues(LeafReaderContext context, BigInteg
     public Object missingObject(Object missingValue, boolean reversed) {
         if (sortMissingFirst(missingValue) || sortMissingLast(missingValue)) {
             final boolean min = sortMissingFirst(missingValue) ^ reversed;
-            return min ? Numbers.MIN_UNSIGNED_LONG_VALUE : Numbers.MAX_UNSIGNED_LONG_VALUE;
+            return min ? Numbers.MIN_UNSIGNED_LONG_VALUE : Numbers.MAX_SIGNED_LONG;
```

Comment:
This is not a right change I believe, we need to find the way to compare missing values using `Long::compareUnsigned`. I will look into it as well shortly, thank you

Reply:
Sure, I have reverted it. Will try using unsigned comparison. Thanks for reviewing 👍

Reply:
Have tried a different approach to safely convert at read time since comparison of long values needs to be changed at Lucene [here](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/comparators/NumericComparator.java#L379)

Reply:
The alternative we may consider is to fallback to `double` based comparison (with precision loss sadly), if changing the comparison logic will be difficult .